src/Profiles/MuscleOxygen/DataPages/Base/ANTPLUS_MuscleOxygenBaseMainDataPageMsg.cpp

```diff
@@ -0,0 +1,52 @@
+#include <Profiles/MuscleOxygen/DataPages/Base/ANTPLUS_MuscleOxygenBaseMainDataPageMsg.h>
+#include <Profiles/MuscleOxygen/ANTPLUS_MuscleOxygenPrivateDefines.h>
+#include <ANTPLUS_PrivateDefines.h>
+
+
+MuscleOxygenBaseMainDataPageMsg::MuscleOxygenBaseMainDataPageMsg(uint8_t dataPageNumber) : BaseDataPageMsg<BroadcastDataMsg>()
+{
+    memset(_buffer, 0, MESSAGE_SIZE);
+    setDataBuffer(_buffer);
+    _buffer[ANTPLUS_DEFAULT_DATAPAGE_BYTE] = dataPageNumber;
+    setCapabilities();
+    setNotifications();
+}
+
+void MuscleOxygenBaseMainDataPageMsg::setEventCount( uint8_t n )
+{
+    _buffer[1] = n;
+}
+
+void MuscleOxygenBaseMainDataPageMsg::setNotifications( uint8_t n )
+{
+    _buffer[2] = n;
+}
+
+void MuscleOxygenBaseMainDataPageMsg::setCapabilities( uint8_t c )
+{
+    _buffer[3] = c;
+}
+
+void MuscleOxygenBaseMainDataPageMsg::setTotalHemoglobinConcentration( uint16_t tc )
+{
+    _buffer[4] = (uint8_t)tc;
+    _buffer[5] &= ~0x0F;
+    _buffer[5] |= (tc >> 8) & 0x0F;
+}
+
+/* TODO
```

Comment:
You already wrote it in your comment below (setPreviousSaturatedHemoglobinPercentage)

Reply:
I think the device just bumps the reading down. So on every event it moves the current value into previous and puts a new current. Docs site this reasoning for in the event of a data loss. It would be awesome if the class could handle this itself but since the class is never preserver I don't think there is a way around it other than showing the user through the example to do it themselves.